src/mumble/MainWindow.cpp

```diff
@@ -954,9 +954,17 @@ void MainWindow::openUrl(const QUrl &url) {
 		}
 	}
 
-	if ((major < 1) ||                                       // No pre 1.2.0
-		(major == 1 && minor <= 1) || (major > thismajor) || // No future version
-		(major == thismajor && minor > thisminor) || (major == thismajor && minor == thisminor && patch > thispatch)) {
+	// We can't handle URLs for versions < 1.2.0
+	const int minMajor = 1;
+	const int minMinor = 2;
+	const int minPatch = 0;
+	const bool isPre_120 = major < minMajor || (major == minMajor && minor < minMinor)
+		|| (major == minMajor && minor == minMinor && patch < minPatch);
+	// We also can't handle URLs for versions newer than the running Mumble instance
```

Comment:
I find this very problematic.

We introduced a new url format with 1.2.0, which introduced the version attribute too because of it.

But as long as we do not change the URL format, we can also open links of newer versions. I do not think we should categorically exclude urls of future versions. That would introduce an unnecessary incompatibility between client versions and make it a lot harder (knowledge and manual work necessary) for users, webmasters and admins to fix up copied urls.

The 1.2.0 version we used effectively worked as a v1 vs v2 url scheme flag. Which is a somewhat unfortunate used label. You could read it as “URL scheme introduced with 1.2.0” though, which works. Still, I would have probably preferred a more generic, less misleading version labeling - for example a simple `v2`.

Reply:
GitHub shows 3 additional lines for context apparently, but this comment is specifically to the code comment (and concept behind it).

Reply:
1. This restriction was not introduced by this PR. Mumble already checks if the version is newer and rejects the URL if it is. I merely refactored the code in order to make it more readable what is being checked for.
2. Given that we are using Mumble versions (it seems?) for identifying the URL scheme, we just can't assume that newer versions did not temper with the URL format. Otherwise we would not have to include the version String into the URL in the first place.
3. I would be fine with using independent versioning for the URLs. In fact I think it makes a lot more sense. That should probably be done in a separate PR though :thinking:

Reply:
Ok, never mind this concern then.